### PR TITLE
Adds preferKernel option to JupyterLab code

### PR DESCRIPTION
**Pull request by jweill-aws**
## References

See #12259, https://github.com/jupyterlab/jupyterlab_server/pull/248, https://github.com/jupyterlab/jupyterlab_server/pull/250.

## Code changes

Modifies the code in JupyterLab to depend on the `preferKernel` option in `PageOptions`.

## User-facing changes

None.

## Backwards-incompatible changes

None. (If option is not set, `true` is assumed.)

---
Edit: Fix issue number

### CODE REVIEW

**Comment by fcollonval** [comment]
@jweill-aws would you mind providing the reason why using a server setting rather than a frontend setting?

**Comment by jweill-aws** [comment]
As discussed in our Wednesday JupyterLab meeting, we will be adding this to the server settings because we intend for administrators to set this option in limited circumstances.

File: packages/notebook-extension/src/index.ts [hunk 1]
```diff
@@ -816,6 +816,14 @@ function activateWidgetFactory(
   translator: ITranslator,
   settingRegistry: ISettingRegistry | null
 ): NotebookWidgetFactory.IFactory {
+  const preferKernelOption = PageConfig.getOption('notebookStartsKernel');
+
+  // If the option is not set, assume `true`
+  const preferKernelValue: boolean =
+    preferKernelOption === ''
+      ? true
+      : preferKernelOption.toLowerCase() === 'true';
```
Comment (fcollonval) [inline packages/notebook-extension/src/index.ts:825]: I think this is clearer to tell it is true if not set or set to `true`.

```suggestion
  const preferKernelValue = preferKernelOption === '' || preferKernelOption.toLowerCase() === 'true';
```

**Review by fcollonval — APPROVED** [review]
I have a small code style suggestion but otherwise this looks good.

Thanks @jweill-aws

**Comment by fcollonval** [comment]
@meeseeksdev please backport to master

> @jweill-aws could you target `master` next time please? It is better (less likely to never show up in the main branch) to merge PRs in master and then to backport them than the other way around.

**Comment by fcollonval** [comment]
@meeseeksdev please backport to master

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.4.x

**Comment by fcollonval** [comment]
Backport to 3.4.x not needed as it was released on 3.3.x branch

**Pull request merged**